doc/syslog/syslog-rate-limit-design.md

```diff
@@ -0,0 +1,367 @@
+
+show syslog rate-limit-container bgp
+SERVICE    INTERVAL     BURST
+--------   ----------   --------
+bgp        500          5000
+```
+
+#### YANG model
+
+```yang
+...
+container SYSLOG_CONFIG {
+
+    description "SYSLOG_CONFIG part of config_db.json";
+
+    container GLOBAL {
+        leaf rate_limit_interval {
+            description "Message rate limit interval";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+
+        leaf rate_limit_burst {
+            description "Message rate limit burst";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+    }
+    /* end of list SYSLOG_CONFIG_LIST */
+}
+/* end of container SYSLOG_CONFIG */
+
+container SYSLOG_CONFIG_FEATURE {
+
+    description "SYSLOG_CONFIG_FEATURE part of config_db.json";
+
+    list SYSLOG_CONFIG_FEATURE_LIST {
+
+        key "service";
+
+        leaf service {
+            description "Service name";
+            type leafref {
+                path "/feature:sonic-feature/feature:FEATURE/feature:FEATURE_LIST/feature:name";
+            }
+        }
+
+        leaf rate_limit_interval {
+            description "Message rate limit interval";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+
+        leaf rate_limit_burst {
+            description "Message rate limit burst";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+    }
+    /* end of list SYSLOG_CONFIG_FEATURE_LIST */
+}
+/* end of container SYSLOG_CONFIG_FEATURE */
+...
+
+
+container FEATURE {
+
+    description "feature table in config_db.json";
+
+    list FEATURE_LIST {
+        ...
+
+        leaf support_syslog_rate_limit {
+            description "This configuration indicates if the feature support configuring syslog rate limit";
+            type stypes:boolean_type;
+            default "false";
+        }
+
+        leaf has_customized_syslog_rate_limit {
+            description "This configuration indicates if the feature has customized syslog rate limit configuration";
+            type stypes:boolean_type;
+            default "false";
+        }
+
+        ...
+    }
+}
+```
+
+### Warmboot and Fastboot Design Impact
+
+N/A
+
+### Restrictions/Limitations
+
+1. Cannot support container not registered to FEATURE table
+2. Persist syslog configuration for database container will not be loaded on next reboot/reload.
+3. Configuring rate limit would cause rsyslogd dropping some log messages because it will restart rsyslogd.
+
+### Testing Requirements/Design
+
+#### Unit Test cases
+1. Verify command "config syslog rate-limit-host"
+2. Verify command "config syslog rate-limit-container"
+3. Verify command "show syslog rate-limit-host"
+4. Verify command "config syslog rate-limit-container"
+
+#### sonic-mgmt Test cases
+
+A new test case shall be added, test flow like below:
+1.	Loop each container
```

Comment:
I think this design also covers host side rate limit setting. Do we need host side rate limit test?